RATapi/models.py

```diff
@@ -125,9 +189,39 @@ def __str__(self):
 
 
 class ContrastWithRatio(RATModel):
-    """Groups together all of the components of the model including domain terms."""
-
-    name: str = Field(default_factory=lambda: "New Contrast " + next(contrast_number), min_length=1)
+    """A group of all of the components of a model, including domain terms.
+
+    Parameters
+    ----------
+    name : str
+        The name of the contrast.
+    data : str
+        The name of the dataset used by the contrast.
+    background : str
+        The name of the background for the contrast.
+    background_action : BackgroundActions
+        Whether the background should be added ('add') or subtracted ('subtract') from the data.
+    bulk_in : str
+        The name of the bulk-in parameter which defines the SLD of the interface between the
+        first layer and the environment.
+    bulk_out : str
+        The name of the bulk-out parameter which defines the SLD of the interface between the last
+        layer and the environment.
+    scalefactor : str
+    resolution : str
+        The name of the instrument resolution for this contrast.
+    resample : bool
+        Whether adaptive resampling should be used for interface microslicing.
+    domain_ratio : str
```

Comment:
The domain ratio weights the reflectivities of the two domain contrasts - see `averageReflectivity.m`